contractcourt/commit_sweep_resolver.go

```diff
@@ -181,7 +194,12 @@ func (c *commitSweepResolver) Resolve() (ContractResolver, error) {
 		return nil, err
 	}
 
+	// Wait up until the CSV expires, unless we also have a CLTV that
+	// expires after.
 	unlockHeight := confHeight + c.commitResolution.MaturityDelay
+	if c.hasCLTV() && c.leaseExpiry > unlockHeight {
+		unlockHeight = c.leaseExpiry
```

Comment:
We'll need to factor in the possibilty that the CSV delay is actually _longer_ than the lease duration. So if the CSV delay is 4032, and the lease duration is 2016, then we actually need to wait for 4032 blocks here.